core/src/main/scala/spinal/core/sim/SpinalSpec.scala

```diff
@@ -0,0 +1,276 @@
+package spinal.core.sim
+
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.{BeforeAndAfterAllConfigMap, ConfigMap}
+import spinal.idslplugin.ValCallback
+
+import spinal.core.{Component, Nameable}
+
+/** Define unit / regression tests for a component */
+abstract class SpinalSpec extends AnyFreeSpec with ValCallback with BeforeAndAfterAllConfigMap {
+
+  override def valCallback[T](ref: T, name: String): T = {
+    ref match {
+      case nameable: Nameable => nameable.setName(name)
+      case _                  =>
+    }
+    ref
+  }
+
+  /** Override it to define the default seed for tests (else seed seed is not fixed)
+    *
+    * It can be overriden from sbt command: `testOnly Myspec -- -Dseed=11`.
+    */
+  val defaultSeed: Option[Int] = None
+
+  // Private else user could modify it in a test, leading to strange behavior.
+  private var seed: Option[Int] = None
+
+  /** Seed can be set with `testOnly MyComponentSpec -- -Dseed=123456789` */
+  final override def beforeAll(configMap: ConfigMap): Unit = {
+    seed = configMap
+      .get("seed")
+      .map(_.asInstanceOf[String].toInt)
+      .orElse(defaultSeed)
+  }
+
+  /** Override it to define the default SpinalSimConfig (else defaults to SimConfig) */
+  val defaultConfig: SpinalSimConfig = SimConfig
```

Comment:
Should be def to avoid strange behavior because of interior mutability (for instance `bench(finalConfig = defaultConfig.withWave` => oops, enabled waves possibly everywhere.

```suggestion
  def defaultConfig: SpinalSimConfig = SimConfig
```